Auth/Auth.Platform.Android/Security/AndroidSecurityCheckFactory.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using Android.Content;
+
+namespace AeroGear.Mobile.Security
+{
+    /// <summary>
+    /// Factory for Android security checks.
+    /// </summary>
+    internal class AndroidSecurityCheckFactory : ISecurityCheckFactory
+    {
+        private readonly Context context;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:AeroGear.Mobile.Security.AndroidSecurityCheckFactory"/> class.
+        /// On Android, checks needs to have a reference to the context.
+        /// </summary>
+        /// <param name="ctx">The application context.</param>
+        public AndroidSecurityCheckFactory(Context ctx)
+        {
+            this.context = ctx;
+        }
+
+        /// <summary>
+        /// Returns an initialized instance of the check identified by the passed in pseudo enumeration.
+        /// </summary>
+        /// <returns>The initialized instance of the check.</returns>
+        /// <param name="type">Type of the check to be instantiated. This must be an instance of <see cref="T:AeroGear.Mobile.Security.SecurityChecks"/></param>
+        public ISecurityCheck create(ISecurityCheckType type)
```

Comment:
I would like to find a way to make this be a static method, but it will be hard to call it from platform agnostic code.  My mind goes toward using an if preprocessor directive, but I don't like those.  I think your implementation is probably the best approach.

Reply:
I thought about that as well, but we can't use preprocessor since platform agnostic code has no dependency on the platform-specific code, so the only way we have to access a platform-specific factory, is to inject the instance in some way, hence the non static method (and, however, I don't like proprocessor directives as well ;) )